### PR TITLE
changes to search option, will change UI after

### PR DESCRIPTION
**Pull request by Barnadottir**

### CODE REVIEW

**Review by psoder — CHANGES_REQUESTED** [review]
The changes, while a step in the right direction, are not polished enough. In my opinion the changes do not contribute anything towards the end product since the way it is currently implemented is not scalable. 

The problems with the current PR that I see are:
- `ArtistList` and `ArtistListRow` should be located in their own folder.
- There should probably be a generic list component that can take both artists, tracks and whatever else is needed. 
- `errorHandler` should probably be located in it's own file.
- `errorHandler` should probably be rewritten to give better error messages. Not all `400` errors are that the search query is empty. 
- Don't put everything in `index.tsx`. Use components instead.

In other words, there's a lot of work that needs to be done before trying to extend the core functionality. When building a house one does not start with the roof.

The stuff that you've copied that was already in `index.tsx` the other components was mostly there as proof of concept and not the way it should be done.
